### PR TITLE
Initial Docusaurus rewrite

### PR DESCRIPTION
**Pull request by hzoo**
preview: https://deploy-preview-1521--babel.netlify.com/

## Shouldn't look at the diff since there's too many files changed, looking for more PRs against it

```sh
cd website && yarn && yarn start
```

## priority/easy

- [x] search works
- [x] blog works: had to change  author_url to authorURL and `<!--truncate-->` comments
- [x] links need highlights/underline (I like how https://reactjs.org/ has it)
- [x] make a subdomain and deploy for old website (branch is `old-site`, url is https://old.babeljs.io/)
- [x] create a UI somewhere linking to the old site if someone really needs it, or for our own reference https://github.com/babel/website/pull/1537
- [ ] main docs - need to redo how readme's are rendered `{% include package_readme.html %}` [@existentialism is working on this right now]
  - link to npm/github/edit
- [ ] versioning of v6 and v7 [@existentialism is working on this right now] https://github.com/babel/website/issues/1261
- [ ] handle redirects (https://github.com/babel/website/blob/master/_redirects) https://github.com/babel/website/issues/1535
- [ ] in `<head>`: should we keep things from https://github.com/babel/website/blob/master/_includes/head.html?
---

## special pages
- [x] /env page should be redirected https://github.com/babel/website/pull/1537
- [x] users page https://github.com/babel/website/issues/1528, https://github.com/babel/website/pull/1529
- [x] repl page https://github.com/babel/website/issues/1525
- [x] videos page https://github.com/babel/website/issues/1522, PR started
- [x] team page https://github.com/babel/website/issues/1523, https://github.com/babel/website/pull/1530
- [x] https://github.com/babel/website/issues/1531 code snippet highlighting is off - can use  https://github.com/babel/website/blob/master/_sass/components/_highlight.scss, https://github.com/babel/website/blob/master/_sass/components/_code.scss
- [ ] redo main page (+ sponsors) https://github.com/babel/website/issues/1538
- [ ] 404 page https://github.com/babel/website/issues/1526, pr https://github.com/babel/website/pull/1540
- [ ] setup page https://github.com/babel/website/issues/1527, pr https://github.com/babel/website/pull/1541
- [ ] https://github.com/babel/website/issues/1531 code snippet has no "try" or "copy" functionality now - check https://github.com/babel/website/blob/master/scripts/clipboard.js, https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.5.13/clipboard.min.js, https://github.com/babel/website/blob/master/scripts/open-in-repl.js

### CODE REVIEW

**Comment by xtuc** [comment]
That is looking nice!

Should we keep our old design or change it by the way?

**Comment by hzoo** [comment]
I don't think we should go out of our way to keep it so maybe both 😛 - vuejs.org, https://reactjs.org/, https://deploy-preview-3718--prettier.netlify.com/ look pretty nice @azz if you have ideas

I guess we just need to get the content (docs) back in working order and just continue to iterate otherwise it will take forever for this to happen and we need it asap for the content not to make it look nice.

**Comment by JoelMarcey** [comment]
@hzoo Hi! Thanks for giving Docusaurus a go. Let us know if you have any questions as you work through the conversion. You can jump into our [Discord channel](https://discord.gg/docusaurus), [file an issue](https://github.com/facebook/Docusaurus/issues) on the Docusaurus repo, or just comment here.

cc: @ericnakagawa

File: _data/users.yml [hunk 1]
```diff
@@ -1,51 +1,67 @@
...
   logo: zendesk.svg
+  pinned: true
 - name: Vimeo
   url: https://vimeo.com
   logo: vimeo.svg
+  pinned: true
 - name: npm
   url: https://www.npmjs.com
   logo: npm.svg
+  pinned: true
```
Comment (Daniel15) [inline _data/users.yml]: Is this change related? If not, could you submit it as a separate PR?

**Comment by hzoo** [comment]
Sorry it I didn't mean to ask to review the code perse since at the top I said it would be impossible to do in a PR. It was more about feedback on the UI and the other things

**Comment by Daniel15** [comment]
oh, lol, I'm not sure how I missed that.

This page doesn't work properly (has some markup showing): https://deploy-preview-1521--babel.netlify.com/docs/en/setup.html

One thing we should ensure is kept is URLs. For example, https://deploy-preview-1521--babel.netlify.com/docs/usage/caveats/ throws a 404 because the URL has changed to https://deploy-preview-1521--babel.netlify.com/docs/en/caveats.html.

If it's impossible to keep exactly the same URLs, then the old URLs should 301 (permanent) redirect to new ones. You can do this with Netlify by using a `_redirects` file: https://www.netlify.com/docs/redirects/

**Comment by hzoo** [comment]
@Daniel15 yeah i'm tracking the pages that don't work in the ^ but I didn't do a good job of explaining haha (setup is one of them)

**Comment by hzoo** [comment]
Ok made a new site on netlify so we can make a new branch to target for the old site: `old-site` and a new subdomain: old.babeljs.io for previous stuff

**Comment by GregBorrelly** [comment]
Hey guys, are there any tasks I can take over? I would love to participate.

**Comment by SavePointSam** [comment]
Same here! I'd love to tackle some thing/s this weekend :D

**Comment by hzoo** [comment]
Yeah @GregBorrelly, @SavePointSam, and anyone else looking - I already made a few issues tagged in this PR that people can do as separate tasks. It's mostly making sure that we can bring back functionality from the current site (babeljs.io) into this one (check the preview url), so anything you can do for that would be appreciated since it all adds up. Sometimes it's worth rewriting too if you feel inspired too.

**Comment by hzoo** [comment]
Will make a new issue https://github.com/babel/website/issues/1485#issuecomment-361727885
